### PR TITLE
Observations permissions

### PR DESCRIPTION
**Pull request by arbianchi**
1) Set up access control for which users to create/read observations:
- Researchers can click 'Observations' button in nav bar then see and create observations
- Scientists can click 'Add Observation' button in nav bar then can only create observations
- No tests

### CODE REVIEW

**Comment by ahirschberg** [comment]
Tests are failing because the test runners don't have authentication for the observation pages

**Comment by jkeam** [comment]
:( failing build

**Comment by jkeam** [comment]
I think you can look into FactoryGirl to create some authenticated users.

File: spec/controllers/observations_controller_spec.rb [hunk 1]
```diff
@@ -43,87 +43,141 @@
...
+    context "no account" do
+      it "allows access" do
+        get :new
+        expect(response).not_to render_template(:new)
+        expect(response).to redirect_to(root_url)
+        
+        # TODO: figure out why the observation is still set, even when
+        # the user does not have permission. By my reasoning, this
+        # assertion should FAIL!
+        expect(assigns(:observation)).to be_a_new(Observation)
```
Comment (ahirschberg) [inline spec/controllers/observations_controller_spec.rb:136]: Why is a new observation successfully assigned, even though the template redirects to root? I couldn't find any of the methods in the controller being called, but commenting out the entire class results in this assertion failing.

File: spec/controllers/observations_controller_spec.rb [hunk 2]
```diff
@@ -43,87 +43,141 @@
...
+    context "scientist account" do
+      it "allows access" do
+        sign_in scientist
+        get :new
+        expect(assigns(:observation)).to be_a_new(Observation)
+        expect(response).to render_template(:new)
+      end
+    end
+    context "no account" do
+      it "allows access" do
```
Comment (ahirschberg) [inline spec/controllers/observations_controller_spec.rb]: Should read "denies access"

**Comment by ahirschberg** [comment]
Yeah I'm still working on getting all the tests to pass, should be done in a day or so.

I also noticed something strange about our prior tests. [There's an assertion passing that I don't think should, and I can't figure out why](https://github.com/rubyforgood/loggerhead_shrike/pull/56/files#diff-2340d5c5e9640d2598d96360d7a1fd37R132). Some of the old tests might be broken but still passing.

**Comment by ahirschberg** [comment]
@jjlangholtz @jkeam this should be ready for a code review. 

Or should we fix all the formatting stuff houndci is freaking out about before merging to master?

**Comment by jjlangholtz** [comment]
No need to worry about the houndci violations right now, good work with the passing tests, looks good to me 👍

**Comment by jkeam** [comment]
👍  🍻

**Comment by jkeam** [comment]
Gonna blow this branch away.  We've got a good number of extra branches I don't think we need anymore hanging around.

**Pull request merged**